### PR TITLE
feat: Homepage

### PR DESCRIPTION
**Pull request by bzgaspar**
Create homepage content. Fetch content using graphql and apollo. Responsive from mobile to desktop

[Preview](https://audiophile-e-commerce-website-git-feature-homepage-itsmenek.vercel.app?_vercel_share=xGnH5nUJjI3qHCqZf25HqxdsR3dVRo57)

### CODE REVIEW

File: src/app/page.tsx [hunk 1]
```diff
@@ -1,3 +1,36 @@
-export default function Home() {
-  return <main></main>
+import Banner from '@/components/Banner/Banner'
+import Categories from '@/components/Categories/Categories'
+import getBanner from '@/services/getBanner'
+import getCategoriesNavigation from '@/services/getCategoriesNavigation'
+
+export default async function Home() {
+  const hero_banner = await getBanner({
```
Comment (eremocal) [inline src/app/page.tsx]: please be consistent with naming variables

File: src/services/getBanner.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    }
+  }
+`
+export default async function getBanner({
+  pageQ,
+  bannerTypeQ,
+}: GetBannerQueryVariables) {
+  const data = await executeApolloQuery({
+    query: QUERY,
+    variables: { pageQ: pageQ, bannerTypeQ: bannerTypeQ },
```
Comment (eremocal) [inline src/services/getBanner.ts]: any reason why the variables has prefix name?

File: src/apollo/ApolloClient.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import {
+  ApolloClient,
+  DocumentNode,
+  InMemoryCache,
+  createHttpLink,
+} from '@apollo/client'
+import { setContext } from '@apollo/client/link/context'
+
+const httpLink = createHttpLink({
+  uri: `https://graphql.contentful.com/content/v1/spaces/${process.env.CONTENTFUL_SPACE_ID}/environments/master`,
```
Comment (eremocal) [inline src/apollo/ApolloClient.tsx]: please put the master to a env

File: src/services/getBanner.ts (hunk 2 above)
Comment (bzgaspar) [reply src/services/getBanner.ts]: Im having error with the query if i use 'page'.

File: src/components/Banner/Banner.tsx [hunk 4]
```diff
@@ -38,29 +45,31 @@ function Banner({ data }: BannerProps) {
...
-              type='button'
-              className={`
+              >
+                {tag}
+              </span>
+            )}
+            <h2>{parseTextWithSpan(title, emphasis)}</h2>
+            {description && <p>{description}</p>}
+            {product && (
+              <Link
```
Comment (eremocal) [inline src/components/Banner/Banner.tsx:62]: do we have a styling for active states?

**Review by eremocal — COMMENTED** [review]
please be consistent when naming variables

File: src/services/getCategoriesNavigation.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    }
+  }
+`
+
+async function getCategoriesNavigation(
+  isCategory?: GetCategoriesQueryVariables
+) {
+  const queryOptions = {
+    query: QUERY_CAT,
+    variables: isCategory ? { isCategory: true } : undefined,
```
Comment (eremocal) [inline src/services/getCategoriesNavigation.ts]: ```...(isCategory && { variables: { isCategory: true } }),```

File: src/components/common/overlay/Overlay.module.scss [hunk 6]
```diff
@@ -1,10 +1,10 @@
 .overlay {
   display: flex;
   position: fixed;
-  top: 5.6rem;
+  top: 0;
   left: 0;
   width: 100%;
-  height: 100vh;
-  z-index: 1;
+  height: 400vh;
```
Comment (eremocal) [inline src/components/common/overlay/Overlay.module.scss]: do you really need 400vh here?

**Review by eremocal — APPROVED** [review]

**Pull request merged**